### PR TITLE
Adds a water reaction to gold slimes + tweaks blacklist.

### PR DESCRIPTION
**Pull request by harmonyn**
Changes in this PR:
- Adds a water reaction to gold slimes. This produces pets and/or other friendly/neutral mobs like parrots and crabs. For when you want your sentience potion army to be less horrifying and more adorable. (Fluff: Blood makes them carnivores, water does not.)
- Removes clowns from the non-friendly gold slime blacklist. If you are creating a game-breaking amount of clowns with gold slimes you probably deserved it, and there are worse things that can spawn anyway. HONK.

### CODE REVIEW

**Comment by KorPhaeron** [comment]
/mob/living/simple_animal/hostile/retaliate is the abstract/generic/parent/whatever its called, it will spawn an invisible mob.

Clowns I assume got left out because they create human bodies or something.

Bats and goats already spawn.

**Comment by Cheridan** [comment]
>fluff

where the hell do you think you are

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 1]
```diff
@@ -50,7 +48,15 @@
 			/mob/living/simple_animal/hostile/blob,
 			/mob/living/simple_animal/ascendant_shadowling
 			)//exclusion list for things you don't want the reaction to create.
-		var/list/critters = typesof(/mob/living/simple_animal/hostile) - blocked // list of possible hostile mobs
+		var/list/meancritters = typesof(/mob/living/simple_animal/hostile) - blocked // list of possible hostile mobs
+		var/list/nicecritters = list(typesof(/mob/living/simple_animal/pet),
```
Comment (Incoming5643) [inline code/modules/reagents/Chemistry-Recipes.dm]: Is this valid syntax? I would think this would create a list within the list of those types(i.e. nicecritters = list(list(pet, cat, dog, pug...), crab, mouse...)

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 2]
```diff
@@ -35,8 +35,6 @@
 			/mob/living/simple_animal/hostile/syndicate/ranged,
 			/mob/living/simple_animal/hostile/syndicate/ranged/space,
 			/mob/living/simple_animal/hostile/alien/queen/large,
-			/mob/living/simple_animal/hostile/retaliate,
-			/mob/living/simple_animal/hostile/retaliate/clown,
```
Comment (Incoming5643) [inline code/modules/reagents/Chemistry-Recipes.dm]: As was said, retaliate is an abstract parent and should be blocked. The blacklist doesn't catch children of listed types, so this set up allowed bats and goats while blocking clowns and abstracts.

There are a few abstracts that do spawn though, like the animated shooty gun.

File: code/modules/reagents/Chemistry-Recipes.dm (hunk 1 above)
Comment (RemieRichards) [inline code/modules/reagents/Chemistry-Recipes.dm]: it's valid syntax for list in list, which is not how these critter lists work. ```pick()``` is not smart enough to choose an element from the embedded list, it just returns the list.

tl;dr this is wrong, it needs to be 
```nicecritters += typesof(/mob/living/simple_animal/pet) - /mob/living/simple_animal/pet```

Comment (bgobandit) [inline code/modules/reagents/Chemistry-Recipes.dm]: is it? do I really have to add all the pets manually fucking hell

File: code/modules/reagents/Chemistry-Recipes.dm (hunk 2 above)
Comment (bgobandit) [inline code/modules/reagents/Chemistry-Recipes.dm]: kk

can I at least remove cluwnes from blacklist though

File: code/modules/reagents/Chemistry-Recipes.dm (hunk 1 above)
Comment (RemieRichards) [inline code/modules/reagents/Chemistry-Recipes.dm]: You don't have to add the pets manually.
you just have to make the ```typesof(/blah/blah/pet)``` be ADDED to the list, instead of as an element of the list.

**Comment by Ergovisavi** [comment]
Clowns should remain on the blacklist. This is because when you spawn hostile gold slime mobs, you want them to actively seek out targets and murder them, not sit around in a hallway honking.

**Comment by bgobandit** [comment]
a) clowns are dangerous as shit if you provoke them, they are more than adequate for murder, certainly more adequate than other shit that spawns

b) not anymore, the primary use for gold slime mobs these days is sentience potion fodder. take it from someone who plays xenobiologist a lot.
